src/operator/quantization/quantize_v2.cc

```diff
@@ -47,6 +47,22 @@ static bool QuantizeV2StorageType(const nnvm::NodeAttrs& attrs, const int dev_ma
   return true;
 }
 
+static OpStatePtr CreateQuantizeV2State(const nnvm::NodeAttrs& attrs, Context ctx,
+                                        const std::vector<TShape>& in_shapes,
+                                        const std::vector<int>& in_types) {
+  OpStatePtr state;
+  if (ctx.dev_type == kGPU) {
+    state = OpStatePtr::Create<QuantizeV2Operator<gpu>>(attrs);
```

Comment:
why are we changing the gpu op to be stateful to ? seems like that doesnt help much.

Reply:
You're right. Making gpu op stateful doesn't bring any help to gpu performance. This is because of the limitation of FW, that is, if an op registered `FCreateOpState`, then it will be treated as stateful op for all backends. `FComputeEx` and `FCompute` will be ignored. Only `FStatefulCompute` and `FStatefulComputeEx` are used. To implement `FStatefulCompute<gpu>`, we have to create a state for it.

Reply:
> This is because of the limitation of FW, that is, if an op registered FCreateOpState, then it will be treated as stateful op for all backends. FComputeEx and FCompute will be ignored. 

maybe i am missing something but seems like this should be fixed in the FW. 

also you could work around this by adding FCreateOpState inside the #if MXNET_USE_MKLDNN == 1 check, seems like neither non-mkldnn cpu or gpu implementation requires stateful op.

Reply:
Your suggestion will have problem to build mkldnn and cuda together, just as the case of mxnet_cu92mkl. for mxnet_cu92mkl, `MXNET_USE_MKLDNN == 1` is enabled, but end-user may use GPU only.

Reply:
yes. makes sense to keep the way it is. hopefully this wont cause any perf impact.